elide-graphql/src/main/java/com/yahoo/elide/graphql/parser/VariableResolver.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2019, Oath Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+
+package com.yahoo.elide.graphql.parser;
+
+import graphql.language.ArrayValue;
+import graphql.language.BooleanValue;
+import graphql.language.EnumValue;
+import graphql.language.FloatValue;
+import graphql.language.IntValue;
+import graphql.language.ListType;
+import graphql.language.NonNullType;
+import graphql.language.NullValue;
+import graphql.language.ObjectField;
+import graphql.language.ObjectValue;
+import graphql.language.OperationDefinition;
+import graphql.language.StringValue;
+import graphql.language.Type;
+import graphql.language.Value;
+import graphql.language.VariableDefinition;
+import graphql.language.VariableReference;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.ws.rs.BadRequestException;
+
+/**
+ * Class that contains variables provided in graphql request and can resolve variables based on
+ * {@link graphql.language.OperationDefinition} scope.
+ */
+class VariableResolver {
```

Comment:
All of the entity projection maker tests will need unit tests or our coverage will drop a lot.